static/js/dialog_widget.js

```diff
@@ -155,3 +156,32 @@ export function launch(conf) {
         on_hidden: conf?.on_hidden,
     });
 }
+
+export function submit_api_request(
+    request_method,
+    url,
+    data = {},
+    {
+        failure_msg_html = $t_html({defaultMessage: "Failed"}),
+        success_continuation,
+        error_continuation,
+    } = {},
+) {
+    request_method({
+        url,
+        data,
+        success(reponse_data) {
+            close_modal();
+            if (success_continuation !== undefined) {
+                success_continuation(reponse_data);
+            }
+        },
+        error(xhr) {
+            ui_report.error(failure_msg_html, xhr, $("#dialog_error"));
+            hide_dialog_spinner();
```

Comment:
This will try to hide the loading spinner even if the modal doesn't make use of it.

Reply:
We are trying to make use of spinner in every modal, so I thought it will be clean.

Reply:
It's harmless to hide the dialog spinner if it was already not shown. This feels a bit unclean, but probably the right way to fix that would be to refactor a bit so that this new function is a class method on a `dialog_widget` class, so we can access whether there is a spinner or not.

Though I think the actual issue is probably that `submit_api_request` should be starting a spinner, right?

Reply:
Merged this with the change to add a loading spinner.